app/full_text_search.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+
+from google.appengine.api import search
+
+import model
+
+# The index name for full text search
+PERSON_FULLTEXT_INDEX_NAME = 'person_information'
+
+def search_with_index(repo, query_obj, max_results):
+    """
+    Searches person with index.
+    Args:
+        repo: The name of repository
+        query_obj: Search word
+        max_results: The max results you want.(Maximum: 1000)
+    Returns:
+        results[<model.Person>, ...]
+    """
+    def create_query(query):
+        query_words = query.split(' ')
+        query_string = ''
+        for word in query_words:
+            if (word != ''):
+                query_string += word + ' AND '
+        return query_string[:-5]
+
+    results = []
+    if (query_obj == ''):
+        return results
+    query = create_query(query_obj)
+    try:
+        index = search.Index(name=PERSON_FULLTEXT_INDEX_NAME)
+        options = search.QueryOptions(
+            limit=max_results,
+            returned_fields=['record_id'])
+        results_index = index.search(search.Query(
+            query_string=query, options=options))
+        record_ids = []
+        for document in results_index:
+            record_ids.append(document.fields[0].value)
+        for id in record_ids:
+            results.append(model.Person.get_by_key_name(repo + ':' + id))
+    except search.Error:
```

Comment:
Do you know when this exception can be thrown? If it happens only in unexpected case such as the index is broken, it would be better to remove this try-except. Then the exception is not caught and cause HTTP 500 error.

If you do so, please document that this method can throw search.Error in the method comment.